tools/flaky-test-reporter/ghutil/fakeghutil/fakeghutil.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// fakeghutil.go fakes GithubClient for testing purpose
+
+package fakeghutil
+
+import (
+	"fmt"
+
+	"github.com/google/go-github/github"
+	"github.com/knative/test-infra/tools/flaky-test-reporter/ghutil"
+)
+
+// FakeGithubClient is a faked client, implements all functions of ghutil.GithubClientInterface
+type FakeGithubClient struct {
+	User     *github.User
+	Repos    []string
+	Issues   map[string][]*github.Issue     // map of repo: slice of issues
+	Comments map[int][]*github.IssueComment // map of issueNumber: slice of comments
+
+	NextNumber int // number to be assigned to next newly created issue/comment
+}
+
+// GetGithubUser gets current authenticated user
+func (fgc *FakeGithubClient) GetGithubUser() (*github.User, error) {
+	return fgc.User, nil
+}
+
+// ListRepos lists repos under org
+func (fgc *FakeGithubClient) ListRepos(org string) ([]string, error) {
+	return fgc.Repos, nil
+}
+
+// ListIssuesByRepo lists issues within given repo, filters by labels if provided
+func (fgc *FakeGithubClient) ListIssuesByRepo(org, repo string, labels []string) ([]*github.Issue, error) {
+	return fgc.Issues[repo], nil
+}
+
+// CreateIssue creates issue
+func (fgc *FakeGithubClient) CreateIssue(org, repo, title, body string) (*github.Issue, error) {
+	issueNumber := fgc.getNextNumber()
```

Comment:
Is getNextNumber() thread/(go routine) safe.

Reply:
The client is created for each test, so there is no risk of race condition